### ISSUE TITLE
DSN defination problem: Cluster hostname

### ISSUE THREAD

**Issue by faruktemur** [snc/SncRedisBundle#267]
Hi @snc ,

We successfuly connect to redis when we define hostname of all 3 nodes in redis cluster.

We are also connecting to redis cluster when we define directly AWS LoadBalancer's hostname to DSN. But, client is not catching MOVED exceptions. As we understand client is not connecting as cluster mode if we define only 1 hostname.

my config.yml

```
snc_redis:
    clients:
        cluster:
            type: predis
            alias: cluster
            logging: "%kernel.debug%"
            dsn: ['redis://example.eu-west-1.elb.amazonaws.com']
            options:
                profile: 2.4
                connection_timeout: 10
                connection_persistent: true
                read_write_timeout: 30
                iterable_multibulk: false
                throw_errors: true
                cluster: "redis-cluster"
                replication: false
```

**Comment by fthyilmz** [comment]
#269

**Comment by snc** [comment]
The cluster option expects a class name or callable (see https://github.com/nrk/predis/wiki/Client-Options). Your config contains `cluster: "redis-cluster"` which looks weird.

**Comment by fthyilmz** [comment]
Hi @snc 

You are passing string to predis client directly without any validation. Thus, "**redis-cluster**" option creates instance of RedisCluster class as you can see in below.

The cluster options: [See](https://github.com/nrk/predis/blob/v1.1/src/Configuration/ClusterOption.php#L35-L49)

This was our hack. 

Actually, our main problem is not creating RedisCluster instance with only one hostname because of statement in predis [See](https://github.com/nrk/predis/blob/v1.1/src/Client.php#L109)

In our application there is a Load Balancer in front of all Redis Cluster. That's why we have added cluster option statement in pull request.

**Comment by ebarault** [comment]
> Actually, our main problem is not creating RedisCluster instance with only one hostname because of statement in predis

@fthyilmz: finally did you find a solution to this problem? i saw you closed your PR #269 

I'm trying to use aws's Redis which comes with a single "configuration endpoint" which is somewhat similar to your load balander in front of your cluster, but i'm having "MOVED" except

**Comment by curry684** [comment]
Fixed by #416
